src/vmm/src/arch/aarch64/mod.rs

```diff
@@ -58,9 +58,12 @@ pub const MMIO_MEM_SIZE: u64 = layout::DRAM_MEM_START - layout::MAPPED_IO_START;
 
 /// Returns a Vec of the valid memory addresses for aarch64.
 /// See [`layout`](layout) module for a drawing of the specific memory model for this platform.
-pub fn arch_memory_regions(size: usize) -> Vec<(GuestAddress, usize)> {
-    let dram_size = min(size, layout::DRAM_MEM_MAX_SIZE);
-    vec![(GuestAddress(layout::DRAM_MEM_START), dram_size)]
+pub fn arch_memory_regions(offset: usize, size: usize) -> Vec<(GuestAddress, usize)> {
+    let dram_size = min(size, layout::DRAM_MEM_MAX_SIZE - offset);
```

Comment:
I hit this as well when writing the pmem PoC. It's mostly a nit, but what should the semantics be if someone passes (`offset`, `size`) that ends up being past `DRAM_MEM_MAX_SIZE`? Should we actually error out (return an error or `panic`) or at the very least print a warning?

Reply:
Ah, yeah... I focussed a lot on the x86 version of this function and kinda forgot about the ARM one. Let me do the same things here I did over on x86 :)